graphics/primitives.cpp

```diff
@@ -436,19 +436,98 @@ void drawEllipse(int x0, int y0, int x1, int y1, uint32 color, bool filled, void
 
 	while (y0-y1 < b) {  /* too early stop of flat ellipses a=1 */
 		if (filled) {
-			drawHLine(x0 - 1, x0 - 1, y0, color, plotProc, data); /* -> finish tip of ellipse */
-			drawHLine(x1 + 1, x1 + 1, y0, color, plotProc, data);
-			drawHLine(x0 - 1, x0 - 1, y1, color, plotProc, data);
-			drawHLine(x1 + 1, x1 + 1, y1, color, plotProc, data);
+			drawHLine(x0 - 1, x0 - 1, y0, color, data); /* -> finish tip of ellipse */
+			drawHLine(x1 + 1, x1 + 1, y0, color, data);
+			drawHLine(x0 - 1, x0 - 1, y1, color, data);
+			drawHLine(x1 + 1, x1 + 1, y1, color, data);
 		} else {
-			(*plotProc)(x0 - 1, y0, color, data); /* -> finish tip of ellipse */
-			(*plotProc)(x1 + 1, y0, color, data);
-			(*plotProc)(x0 - 1, y1, color, data);
-			(*plotProc)(x1 + 1, y1, color, data);
+			drawPoint(x0 - 1, y0, color, data); /* -> finish tip of ellipse */
+			drawPoint(x1 + 1, y0, color, data);
+			drawPoint(x0 - 1, y1, color, data);
+			drawPoint(x1 + 1, y1, color, data);
 		}
 		y0++;
 		y1--;
 	}
 }
 
+class DeprecatedPrimitives final : public Primitives {
```

Comment:
@ccawley2011 do we actually need this now? Why not remove these methods altogether?